src/lib/ptree/json.lua

```diff
@@ -86,6 +103,50 @@ local function read_json_string(input)
    return table.concat(parts)
 end
 
+local function read_json_array(input)
+   skip_whitespace(input)
+   consume(input, "[")
+   skip_whitespace(input)
+   local ret = {}
+   if not check(input, "]") then
+      repeat
+         skip_whitespace(input)
+         local v = read_json_type(input)
+         skip_whitespace(input)
+         table.insert(ret, v)
+      until not check(input, ",")
+      skip_whitespace(input)
+      consume(input, "]")
+   end
+   skip_whitespace(input)
+   return ret
+end
+
+local function read_json_scalar(input)
+   skip_whitespace(input)
+   local v = take_while(input, '[0-9%-%.truefalsn]')
+   skip_whitespace(input)
+   if constants[v] ~= nil then
+      return constants[v]
+   end
+   local num = tonumber(v)
+   if num then return num end
+   error('unparseable json number or boolean: '..v)
+end
+
+read_json_type = function(input)
```

Comment:
Can we rename this to simply `read_json` ?  Likewise the normal form of `f = function(...) ... end` is `function f(...) ... end`.